cheatsheets/Secure_Product_Design_Cheat_sheet.md

```diff
@@ -0,0 +1,92 @@
+   5. Least privilege: Use the principle of the least privilege when writing code, such that the code and the system it runs on are given the minimum access rights necessary to perform their functions.
+   6. Secure memory management: Use high-level languages recommended in your Golden Path / Paved Road documentation or properly manage memory to prevent memory-related vulnerabilities such as buffer overflow and use-after-free.
+   7. Avoiding hardcoded secrets: Hardcoded secrets such as passwords and encryption keys should be avoided in the code and should be stored in a secure storage.
+   8. Security testing: Test the software for security vulnerabilities during development and before deployment.
+   9. Auditing and reviewing the code: Regularly audit and review the code for security vulnerabilities, such as by using automated tools or having a third party review the code.
+   10. Keeping up-to-date: Keep up-to-date with the latest security best practices and vulnerabilities to ensure that the software is as secure as possible.
+
+Ensure that you integrate plausibility checks at each tier of your application (from frontend to backend) and ensure that you write unit and integration tests to validate that all critical flows are resistant to the results of [Threat Modelling](Threat_Modeling_Cheat_Sheet.md). Use that to compile use-cases and misuse-cases for each tier of your application.
+
+### 5. Configuration
+
+Building an application securely can all too easily be undone if it's not securely configured. At a minimum we should ensure the following:
+
+1. Bearing in mind the principal of Least Privilege: Limit the access and permissions of system components and users to the minimum required to perform their tasks.
+2. Remembering Defense in Depth: Implement multiple layers of security controls to protect against a wide range of threats.
+3. Ensuring Secure by Default: Configure systems and software to be secure by default, with minimal manual setup or configuration required.
+4. Secure Data: Protect sensitive data, such as personal information and financial data, by encrypting it in transit and at rest. Protecting that data also means ensuring it's correctly backed up and that the data retention is set correctly for the desired Product Security Level.
+5. Plan to have the configuration Fail Securely: Design systems to fail in a secure state, rather than exposing vulnerabilities when they malfunction.
+6. Always use Secure Communications: Use secure protocols for communication, such as HTTPS, to protect against eavesdropping and tampering.
+7. Perform Regular updates - or leverage maintained images: Keeping software, docker images and base operating systems up-to-date with the latest security patches is an essential part of maintaining a secure system.
```

Comment:
```suggestion
7. Perform regular updates - or leverage maintained images: Keeping software, docker images and base operating systems up-to-date with the latest security patches is an essential part of maintaining a secure system.
```

Reply:
Seems like # 7 could use a reference to best-practices. Not sure we have a CS for that, but surely OWASP or NIST has something we could reference.

Reply:
NIST does for patch management and I've linked CIS for hardened images now too